lib/hmr/lazyCompilationBackend.js

```diff
@@ -5,20 +5,28 @@
 
 "use strict";
 
-const http = require("http");
-
+/** @typedef {import("tls").SecureContextOptions} SecureContextOptions */
 /** @typedef {import("../Compiler")} Compiler */
 
+/**
+ * @typedef {Object} BackendOptions
+ * @property {SecureContextOptions?} https HTTPS configuration passed straight to Node.js HTTPS module.
+ * @property {number?} port Custom port for lazy compilation backend. If not defined, random port will be used.
+ */
+
 /**
  * @param {Compiler} compiler compiler
  * @param {string} client client reference
  * @param {function(Error?, any?): void} callback callback
+ * @param {?BackendOptions} options additional options for the backend
  * @returns {void}
  */
-module.exports = (compiler, client, callback) => {
+module.exports = (compiler, client, callback, options) => {
 	const logger = compiler.getInfrastructureLogger("LazyCompilationBackend");
 	const activeModules = new Map();
 	const prefix = "/lazy-compilation-using-";
+	options = typeof options === "object" ? options : null;
+	const protocol = options && options.https ? "https" : "http";
```

Comment:
It should be better and do not really not `https` option,  if options contains `key` or `cert` we should run https, otherwise http and pass all allowed options

Reply:
A lot of this is based on webpack-dev-server's behavior.

Reply:
https://github.com/webpack/webpack-dev-server/blob/master/lib/Server.js#L1529

I understand what you're suggesting, and can totally do that. I was mimicking the behavior found on the webpack-dev-server where `https` is passed as the options down to the `createServer`.

Reply:
Yes, we are working on refactor right now, top https and port and enc options is like sugar for developers, under the hood we will have `this.options.server` and `this.options.webSocketServer` options, so `https` and `port  options will be mvoed in `this.options.server` options, like we have `client` and `backed` here

Reply:
Thanks for the clarification. I suppose the property should then just be `serverOptions`?

Reply:
Yes, backendOptions === serverOptions